python/config.py

```diff
@@ -0,0 +1,27 @@
+class config_error(Exception):
+    def __init__(self, sec, key, val, msg, index=0):
+        cache = str(sec)+"."+str(key)
+        if index: cache+="["+str(index)+"]"
```

Comment:
```suggestion
        if index != 0:
            cache += "[" + str(index) + "]"
```

Slight formatting suggestion: I recommend making the index check explicit, and putting spaces between all operators. Additionally, I'd suggest a new line after every `if`, even if it is short, to promote readability.